src/OptimalTransport.jl

```diff
@@ -672,4 +675,162 @@ function quadreg(mu, nu, C, ϵ; θ=0.1, tol=1e-5, maxiter=50, κ=0.5, δ=1e-5)
     return sparse(γ')
 end
 
+## 1-Dimensional Optimal Transport Functions
+
+"""
+optimal_transport_cost(c,mu::ContinuousUnivariateDistribution,nu::UnivariateDistribution; plan=nothing)
+
+Calculates the optimal transport cost between μ to ν, where
+they are 1-Dimensional distributions and the cost
+function is of the form ``c(x,y) = h(|x-y|)`` such that
+``h`` is a convex function.
+
+Parameters:\n
+c: The cost function, which should be of the form ``c(x,y) = h(abs(x-y))``
+where `h` is a convex function.
+mu: 1-D distribution (e.g. `mu = Distributions.Normal(0,1)`)\n
+nu: 1-D distribution (e.g. `nu = Distributions.Normal(0,1)`)\n
+plan: Optional parameter in case the optimal transport plan is already known. Providing
+it can speed up calculations.
+"""
+function optimal_transport_cost(
+    c, μ::ContinuousUnivariateDistribution, ν::UnivariateDistribution; plan=nothing
+)
+    if plan === nothing
+        g(μ, ν, x) = c(quantile(μ, x), quantile(ν, x))
+        f(x) = g(μ, ν, x)
+        return quadgk(f, 0, 1)[1]
+    else
+        cost = quadgk(0, 1) do q
+            x = quantile(μ, q)
+            return c(x, plan(x))
+        end
+        return cost[1]
+    end
+end
+
+"""
+optimal_transport_plan(c,mu::ContinuousUnivariateDistribution,nu::UnivariateDistribution)
+
+Calculates the pptimal transport plan between μ to ν, where
+they are 1-Dimensional distributions and the cost
+function is of the form ``c(x,y) = h(|x-y|)`` such that
+``h`` is a convex function.
+
+Parameters:\n
+c: The cost function, which should be of the form ``c(x,y) = h(|x-y|)``
+where ``h`` is a convex function.
+μ: 1-D distribution (e.g. `μ = Distributions.Normal(0,1)`)\n
+ν: 1-D distribution (e.g. `μ = Distributions.Normal(0,1)`)\n
+
+Returns the optimal transport plan as a function
+  ```math
+  T(x)=F_\\nu^{-1}(F_\\mu(x)).
+  ```
+"""
+function optimal_transport_plan(
+    c, μ::ContinuousUnivariateDistribution, ν::UnivariateDistribution
+)
+    # Use T instead of γ to indicate that this is a Monge map.
+    T(x) = Distributions.quantile(ν, Distributions.cdf(μ, x))
```

Comment:
No need to qualify the functions:
```suggestion
    T(x) = quantile(ν, cdf(μ, x))
```